msgpack/include/adaptor.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef ARK_MSGPACK_ADAPTOR_HPP
+#define ARK_MSGPACK_ADAPTOR_HPP
+
+#include <Ark/Module.hpp>
+#include <msgpack.hpp>
+
+namespace msgpack
+{
+	MSGPACK_API_VERSION_NAMESPACE(MSGPACK_DEFAULT_API_NS)
+	{
+		namespace adaptor
+		{
+			// convert ark value
+			template<> struct convert<Value>
+			{
+				msgpack::object const& operator()(msgpack::object const& o, Value& v) const
+				{
+					if(o.type == type::NIL)
+					{
+						v = Ark::Nil;
+					}
+					else if(o.type == type::BOOLEAN)
+					{
+						if(o.via.boolean == true)
+							v = Ark::True;
+						else
+							v = Ark::False;
+					}
+					else if(o.type == type::FLOAT64)
+					{
+						v = Value(o.via.f64);
+					}
+					else if(o.type == type::STR)
+					{
+						v = Value(std::string {o.via.str.ptr});
+					}
+					else
+					{
+						Value lst_v {ValueType::List};
+
+						for(unsigned i {0}; i < o.via.array.size; ++ i)
+						{
+							Value ark_object_v;
+
+							if(o.via.array.ptr[i].type == type::NIL)
+							{
+								ark_object_v = Ark::Nil;
+							}
+							else if(o.via.array.ptr[i].type == type::BOOLEAN)
+							{
+								if(o.via.array.ptr[i].via.boolean == true)
+									ark_object_v = Ark::True;
+								else
+									ark_object_v = Ark::False;
```

Comment:
can be summarized as ark_object_v = o.via.array.ptr[i].via.boolean ? True : False;